File: src/OS/domain/event.py
from dataclasses import dataclass, field
from datetime import datetime, timezone
from typing import Any
from uuid import UUID, uuid4


@dataclass
class Event:
    """
    runtime事件对象
    """

    type: str
    payload: dict[str, Any]
    id: UUID = field(default_factory=uuid4)
    created_at: datetime = field(default_factory=lambda: datetime.now(timezone.utc))

File: runtime/domain/event.go
package domain

import "time"

//发给agent的消息
type Event struct {
	ID        ID             `json:"id"`
	Type      string         `json:"type"`
	Payload   map[string]any `json:"payload"`
	CreatedAt time.Time      `json:"created_at"`
}

//创建消息，生成id和时间
func NewEvent(eventType string, payload map[string]any) Event {
	return Event{
		ID:        mustNewID(),
		Type:      eventType,
		Payload:   payload,
		CreatedAt: time.Now().UTC(),
	}
}
